pkg/objectstore/bucket.go

```diff
@@ -131,38 +131,40 @@ func (p *provider) ListBuckets(ctx context.Context) (map[string]Bucket, error) {
 // For safety, does not delete buckets with contents. Caller should ensure
 // that bucket is empty.
 func (p *provider) DeleteBucket(ctx context.Context, bucketName string) error {
-	location, err := getStowLocation(ctx, p.config, p.secret, "")
+	location, err := getStowLocation(ctx, p.config, p.secret)
 	if err != nil {
 		return err
 	}
 	return location.RemoveContainer(bucketName)
 }
 
-func (p *provider) getOrCreateBucket(ctx context.Context, bucketName, region string) (Bucket, error) {
+func (p *provider) getOrCreateBucket(ctx context.Context, bucketName string) (Bucket, error) {
 	d, err := p.GetBucket(ctx, bucketName)
 	if err == nil {
 		return d, nil
 	}
 	// Attempt creating it
-	return p.CreateBucket(ctx, bucketName, region)
+	return p.CreateBucket(ctx, bucketName)
 }
 
 type s3Provider struct {
 	*provider
 }
 
 func (p *s3Provider) GetBucket(ctx context.Context, bucketName string) (Bucket, error) {
-	hostEndPoint := p.hostEndPoint
-	var region string
-	if hostEndPoint == "" {
+	region := p.config.Region
+	if region == "" {
 		var err error
 		region, err = p.getRegionForBucket(ctx, bucketName)
 		if err != nil {
 			return nil, errors.Wrapf(err, "could not get region for bucket %s", bucketName)
 		}
+	}
```

Comment:
Do you need to set `p.config.Region = region` here?

Reply:
Yes, thank you. This should probably be hoisted to the provider's initialization so we don't need to do it every time